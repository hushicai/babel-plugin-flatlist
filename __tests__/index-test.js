const plugin = require('../');
const pluginTester = require('babel-plugin-tester');

const tests = [
  {
    title: 'both `StyleSheet.create` and `module.exports`',
    code: `
const styles = StyleSheet.create({
  verticallyInverted: {
    transform: [{scaleY: -1}],
  },
  horizontallyInverted: {
    transform: [{scaleX: -1}],
  },
});

module.exports = VirtualizedList;
`,
    snapshot: true
  },
  {
    title: 'both `StyleSheet.create` and `export default`',
    code: `
const styles = StyleSheet.create({
  verticallyInverted: {
    transform: [{scaleY: -1}],
  },
  horizontallyInverted: {
    transform: [{scaleX: -1}],
  },
});

export default VirtualizedList;
`,
    snapshot: true
  },
  {
    title: 'both `StyleSheet.create` and `module.exports`, but not sibling',
    code: `
const styles = StyleSheet.create({
  verticallyInverted: {
    transform: [{scaleY: -1}],
  },
  horizontallyInverted: {
    transform: [{scaleX: -1}],
  },
});

const a = 1;

module.exports = VirtualizedList;
`,
    snapshot: true
  },
  {
    title: 'only `StyleSheet.create`',
    code: `
const styles = StyleSheet.create({
  verticallyInverted: {
    transform: [{scaleY: -1}],
  },
  horizontallyInverted: {
    transform: [{scaleX: -1}],
  },
});
`,
    snapshot: true
  },
  {
    title: 'styles object and `module.exports`',
    code: `
const styles = {
  verticallyInverted: {
    transform: [{scaleY: -1}],
  },
  horizontallyInverted: {
    transform: [{scaleX: -1}],
  },
};

module.exports = VirtualizedList;
`,
    snapshot: true
  },
  {
    title: 'styles primitive and `module.exports`',
    code: `
const styles = 1;

module.exports = VirtualizedList;
`,
    snapshot: false
  }
];

pluginTester({
  plugin,
  tests
});
