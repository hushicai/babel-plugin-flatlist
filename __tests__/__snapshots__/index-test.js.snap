// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rewrite VirtualizedList perspective both \`StyleSheet.create\` and \`module.exports\`, but not sibling: both \`StyleSheet.create\` and \`module.exports\`, but not sibling 1`] = `
"
const styles = StyleSheet.create({
  verticallyInverted: {
    transform: [{scaleY: -1}],
  },
  horizontallyInverted: {
    transform: [{scaleX: -1}],
  },
});

const a = 1;

module.exports = VirtualizedList;

      ↓ ↓ ↓ ↓ ↓ ↓

const styles = StyleSheet.create({
  verticallyInverted: {
    transform: [{ scaleY: -1 }]
  },
  horizontallyInverted: {
    transform: [{ scaleX: -1 }]
  }
});

const a = 1;

module.exports = VirtualizedList;
"
`;

exports[`Rewrite VirtualizedList perspective both \`StyleSheet.create\` and \`module.exports\`: both \`StyleSheet.create\` and \`module.exports\` 1`] = `
"
const styles = StyleSheet.create({
  verticallyInverted: {
    transform: [{scaleY: -1}],
  },
  horizontallyInverted: {
    transform: [{scaleX: -1}],
  },
});

module.exports = VirtualizedList;

      ↓ ↓ ↓ ↓ ↓ ↓

const styles = StyleSheet.create({
  verticallyInverted: {
    transform: [{
      scaleY: -1
    }, {
      perspective: 1
    }]
  },
  horizontallyInverted: {
    transform: [{
      scaleX: -1
    }, {
      perspective: 1
    }]
  }
});

module.exports = VirtualizedList;
"
`;

exports[`Rewrite VirtualizedList perspective only \`StyleSheet.create\`: only \`StyleSheet.create\` 1`] = `
"
const styles = StyleSheet.create({
  verticallyInverted: {
    transform: [{scaleY: -1}],
  },
  horizontallyInverted: {
    transform: [{scaleX: -1}],
  },
});

      ↓ ↓ ↓ ↓ ↓ ↓

const styles = StyleSheet.create({
  verticallyInverted: {
    transform: [{ scaleY: -1 }]
  },
  horizontallyInverted: {
    transform: [{ scaleX: -1 }]
  }
});
"
`;

exports[`Rewrite VirtualizedList perspective styles object and \`module.exports\`: styles object and \`module.exports\` 1`] = `
"
const styles = {
  verticallyInverted: {
    transform: [{scaleY: -1}],
  },
  horizontallyInverted: {
    transform: [{scaleX: -1}],
  },
};

module.exports = VirtualizedList;

      ↓ ↓ ↓ ↓ ↓ ↓

const styles = {
  verticallyInverted: {
    transform: [{ scaleY: -1 }]
  },
  horizontallyInverted: {
    transform: [{ scaleX: -1 }]
  }
};

module.exports = VirtualizedList;
"
`;
